stablehlo/dialect/StablehloOps.td

```diff
@@ -2758,6 +2758,16 @@ def StableHLO_ReverseOp: StableHLO_Op<"reverse",
     $operand `,` `dims` `=` $dimensions
       attr-dict `:` custom<SameOperandsAndResultType>(type($operand), type($result))
   }];
+
+  let extraClassDeclaration = [{
```

Comment:
Looks like all other ops implementing this method declare it via the `InferTensorTypeWithReify` trait. Can look at `ClampOp` for example which seems to use a similar impl, let's base the implementation off that, could also use a non-conversion-based test:


1. Use trait to declare (either [`StableHLO_ShapedInterfaceOp`](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/dialect/StablehloOps.td#L1954) or [`InferTensorTypeWithReify`](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/dialect/StablehloOps.td#L1538))
2. Implement in [`StablehloOps.cpp`](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/dialect/StablehloOps.cpp#L1093)
3. Test in [`infer_stablehlo.mlir`](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/tests/infer_stablehlo.mlir#L259)

Reply:
sounds good. done. thanks

Reply:
Thanks!
The option [StableHLO_ShapedInterfaceOp](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/dialect/StablehloOps.td#L1954) is recommended in this case.  [InferTensorTypeWithReify](https://github.com/openxla/stablehlo/blob/ec4ec7847908b06ed6170c4aba96c164a22f7396/stablehlo/dialect/StablehloOps.td#L1538) would have added a specialization of `ReverseOp::inferReturnTypes` which is not need in this case.